### ISSUE TITLE
Comment at end of block before closing } trigger padded-block 'never' rule.

### ISSUE THREAD

**Issue by wraithan** [eslint/eslint#2336]
It appears when you have a comment at the end of a block it is still triggering the `padded-block` rule when set to `'never'`. This is on latest published at time of this issue (v0.19.0). I've included the output, the example, the version command, and our eslintrc below. Let me know if any more information would be helpful.

Command line output:
```
eslint test.js

test.js
  6:1  error  Block must not be padded by blank lines  padded-blocks
```

Example JS (test.js):
```
a()

function a () {
  a()
  // stuff
}
```

eslint version verification:
```
eslint --version
v0.19.0
```

Our .eslintrc:
```
{
  "env": {
    "node": true,
    "browser": false
  },
  "global": [
    "after",
    "afterEach",
    "before",
    "beforeEach",
    "describe",
    "it"
  ],
  "rules": {
    "brace-style": 2,
    "comma-style": [2, "last"],
    "consistent-return": 0,
    "curly": 0,
    "eol-last": 2,
    "eqeqeq": 2,
    "camelcase": 0,
    "dot-notation": 2,
    "func-names": 2,
    "guard-for-in": 2,
    "max-len": [2, 90],
    "max-nested-callbacks": [2, 3],
    "max-params": [2, 5],
    "new-cap": 2,
    "no-comma-dangle": 0,
    "no-console": 1,
    "no-else-return": 2,
    "no-floating-decimal": 2,
    "no-lonely-if": 2,
    "no-mixed-requires": 2,
    "no-multiple-empty-lines": 2,
    "no-new": 2,
    "no-shadow": 1,
    "no-undef": 2,
    "no-underscore-dangle": 0,
    "no-unused-vars": 2,
    "no-use-before-define": 0,
    "padded-blocks": [2, "never"],
    "radix": 2,
    "semi": [2, "never"],
    "space-after-function-name": 0,
    "space-after-keywords": 2,
    "space-before-blocks": 2,
    "space-infix-ops": 2,
    "spaced-line-comment": 2,
    "space-return-throw-case": 2,
    "space-unary-ops": 2,
    "strict": 0,
    "quotes": [0, "single"],
    "wrap-iife": 2
  }
}
```
Past similar issues: https://github.com/eslint/eslint/issues/1938

**Comment by lo1tuma** [comment]
This could probably be bug in espree, but I’m not 100% sure how the comment attachment algorithm is supposed to work.

The problem with your example is, that it relies on ASI and the trailing comment is not attached to the `ExpressionStatement`.

**Comment by nzakas** [comment]
Comment attachment does get confused by ASI - I'm not sure if that's a bug or a feature. :)

**Comment by feross** [comment]
Thanks for attempting a fix @lo1tuma! There are still some cases that fail:

- https://github.com/feross/buffer/blob/5b6235dac2f12217bf0be963a122d8be47855cde/index.js#L502-L503
- https://github.com/feross/webtorrent/blob/459602759e704db7a12586970f8e7498f50e8ab3/bin/cmd.js#L401-L402
- https://github.com/feross/webtorrent/blob/459602759e704db7a12586970f8e7498f50e8ab3/lib/torrent.js#L477-L478
- https://github.com/npm/fstream/blob/6e78ceeee08950bfa358fe2ae3f5801608d0b963/lib/reader.js#L107-L108

**Comment by feross** [comment]
Sorry, didn't realize this wasn't in `master`. Nevermind!

**Comment by feross** [comment]
I'm willing to put $10 on this issue.

There's no BountySource link for some reason (perhaps it's another BountySource bug), but I can pay you directly through Paypal or Bitcoin.

**Comment by LinusU** [comment]
I'm willing to chip in $10 as well, I can pay with Bitcoin.

**Comment by nzakas** [comment]
Sorry, the Bountysource plugin is broken. I've contacted them about it but haven't gotten any useful information.

I'd ideally like to solve this in the parser, but if this is a big pain point, I think @lo1tuma had a solution that didn't require a parser change. We can use that info the parser is fixed to eliminate the pain

**Comment by feross** [comment]
@lo1tuma Is your solution still viable? If it's not too onerous, it might be nice to use it until there's a chance to fix the parser, as @nzakas said.

**Comment by lo1tuma** [comment]
@feross My [proposed solution](https://github.com/eslint/eslint/pull/2379) is a workaround which probably decreases the performance of the rule. I think it would also fix #2788 (I can add a test to verify this). If we want to merge this until the comment attachment algorithm is fixed in espree we should file a new issue to remove the workaround in the future.

**Comment by nzakas** [comment]
Yeah, let's do that. I'm not sure when I'll be up to digging into the comment attachment stuff, it's mind-bending and my mind isn't too flexible ATM.

**Comment by feross** [comment]
@lo1tuma, where should I send the bounty I promised for this issue? Paypal, Bitcoin, Venmo works.

cc @LinusU

**Comment by LinusU** [comment]
I'm on! Bitcoin (or BountySource) would work best for me :fireworks:

**Comment by lo1tuma** [comment]
@feross, @LinusU: Thanks, but I’m not interested in earning money with my contributions, so feel free to keep to money or donate it directly to the [eslint team via BountySource](https://www.bountysource.com/teams/eslint).

**Comment by feross** [comment]
@lo1tuma Donated to the eslint team on BountySource - thanks!

**Comment by LinusU** [comment]
![screen shot 2015-07-15 at 23 44 05](https://cloud.githubusercontent.com/assets/189580/8710628/6f59e710-2b4b-11e5-9619-329b7f20f40b.png)
:+1:
